lib/performance_platform.rb

```diff
@@ -11,56 +11,49 @@ def initialize(token = nil)
       @headers = { 'Authorization' => "Bearer #{token}",
                    'Content-Type' => 'application/json' }
     end
-  end
 
-  class TransactionsByChannel < Base
-    def submit_transactions(count, timestamp = Time.zone.now.iso8601, period = 'day')
+    def submit(data, timestamp = Time.zone.now.iso8601, period = 'day')
       HTTParty.post("#{PP_DOMAIN}#{transaction_endpoint}",
-                    body: data(timestamp, period, count),
+                    body: data_map(timestamp, period, data),
                     headers: headers)
     end
+  end
 
+  class TransactionsByChannel < Base
     private
 
     def transaction_endpoint
       '/data/teaching-jobs-job-listings/transactions-by-channel'
     end
 
-    def data(timestamp, period, count)
+    def data_map(timestamp, period, data)
       { _timestamp: timestamp,
         service: 'teaching_jobs_listings',
         channel: 'digital',
-        count: count,
+        count: data,
         dataType: 'transactions-by-channel',
         period: period }.to_json
     end
   end
 
   class UserSatisfaction < Base
-    def submit(rating_counts, timestamp = Time.zone.now.iso8601, period = 'day')
-      HTTParty.post("#{PP_DOMAIN}#{transaction_endpoint}",
-                    body: data(timestamp, period, rating_counts.values.inject(:+), rating_counts),
-                    headers: headers)
-    end
-
     private
 
     def transaction_endpoint
       '/data/teaching-jobs-job-listings/user-satisfaction'
     end
 
-    def data(timestamp, period, total, rating_counts)
-      data = { _timestamp: timestamp,
-               rating_1: rating_counts[1],
-               rating_2: rating_counts[2],
-               rating_3: rating_counts[3],
-               rating_4: rating_counts[4],
-               rating_5: rating_counts[5],
-               service: 'teaching_jobs_listings',
-               total: total,
-               dataType: 'user-satisfaction',
-               period: period }.to_json
-      data
+    def data_map(timestamp, period, data)
+      { _timestamp: timestamp,
+        rating_1: data[1],
+        rating_2: data[2],
+        rating_3: data[3],
+        rating_4: data[4],
+        rating_5: data[5],
+        service: 'teaching_jobs_listings',
+        total: data.values.inject(:+),
```

Comment:
Similar to above, we know what the data is here. Can we name it `rating_counts` or similar to make that clear?

Reply:
👍